t/signature_details.t

```diff
@@ -0,0 +1,459 @@
+		'PPI::Token::Word',           'sub',
+		'PPI::Token::Word',           'foo',
+		'PPI::Structure::Signature',  '($thing, @)',
+		'PPI::Token::Structure',      '(',
+		'PPI::Statement::Expression', '$thing, @',
+		'PPI::Token::Symbol',         '$thing',
+		'PPI::Token::Operator',       ',',
+		'PPI::Token::Symbol',         '@',
+		'PPI::Token::Structure',      ')',
+		'PPI::Structure::Block',      '{}',
+		'PPI::Token::Structure',      '{',
+		'PPI::Token::Structure',      '}',
+	  ],
+	  "nameless slurpy argument";
+}
+
+SLURPY_HASH_ARGUMENT: {
+	test_document
+	  [ feature_mods => { signatures => 1 } ],
+	  <<'END_PERL',
+		sub foo ($filter, %inputs) {}
+END_PERL
+	  [
+		'PPI::Statement::Sub',        'sub foo ($filter, %inputs) {}',
+		'PPI::Token::Word',           'sub',
+		'PPI::Token::Word',           'foo',
+		'PPI::Structure::Signature',  '($filter, %inputs)',
+		'PPI::Token::Structure',      '(',
+		'PPI::Statement::Expression', '$filter, %inputs',
+		'PPI::Token::Symbol',         '$filter',
+		'PPI::Token::Operator',       ',',
+		'PPI::Token::Symbol',         '%inputs',
+		'PPI::Token::Structure',      ')',
+		'PPI::Structure::Block',      '{}',
+		'PPI::Token::Structure',      '{',
+		'PPI::Token::Structure',      '}',
+	  ],
+	  "slurpy hash argument";
+}
+
+NAMELESS_SLURPY_HASH_ARGUMENT: {
+	test_document
+	  [ feature_mods => { signatures => 1 } ],
+	  <<'END_PERL',
+		sub foo ($thing, %) {}
+END_PERL
+	  [
+		'PPI::Statement::Sub',        'sub foo ($thing, %) {}',
+		'PPI::Token::Word',           'sub',
+		'PPI::Token::Word',           'foo',
+		'PPI::Structure::Signature',  '($thing, %)',
+		'PPI::Token::Structure',      '(',
+		'PPI::Statement::Expression', '$thing, %',
+		'PPI::Token::Symbol',         '$thing',
+		'PPI::Token::Operator',       ',',
+		'PPI::Token::Symbol',         '%',
+		'PPI::Token::Structure',      ')',
+		'PPI::Structure::Block',      '{}',
+		'PPI::Token::Structure',      '{',
+		'PPI::Token::Structure',      '}',
+	  ],
+	  "nameless slurpy hash argument";
+}
+
+EMPTY_SIGNATURE: {
+	test_document
+	  [ feature_mods => { signatures => 1 } ],
+	  <<'END_PERL',
+		sub foo () {}
+END_PERL
+	  [
+		'PPI::Statement::Sub',       'sub foo () {}',
+		'PPI::Token::Word',          'sub',
+		'PPI::Token::Word',          'foo',
+		'PPI::Structure::Signature', '()',
+		'PPI::Token::Structure',     '(',
+		'PPI::Token::Structure',     ')',
+		'PPI::Structure::Block',     '{}',
+		'PPI::Token::Structure',     '{',
+		'PPI::Token::Structure',     '}',
+	  ],
+	  "empty signature";
+}
+
+PROTOTYPE_SIGNATURE: {
+	test_document
+	  [ feature_mods => { signatures => 1 } ],
+	  <<'END_PERL',
+		sub foo :prototype($$) ($left, $right) {}
+END_PERL
+	  [
+		'PPI::Statement::Sub',   'sub foo :prototype($$) ($left, $right) {}',  #
+		'PPI::Token::Word',      'sub',
+		'PPI::Token::Word',      'foo',
+		'PPI::Token::Operator',  ':',
+		'PPI::Token::Attribute', 'prototype($$)',
```

Comment:
out of scope note: as someone who writes a code using `PPI` I will appreciate `PPI::Token::Attribute::Prototype` with `@ISA = qw (PPI::Token::Attribute PPI::Token::Token::Prototype)`